typedoc.json

```diff
@@ -0,0 +1,10 @@
+{
+  "entryPoints": ["src/index.ts"],
+  "out": "docs/api",
+  "name": "vexflow",
+  "excludeProtected": true,
+  "excludePrivate": true,
+  "disableSources": true,
+  "hideGenerator": true,
+  "intentionallyNotExported": ["OffscreenCanvas", "OffscreenCanvasRenderingContext2D"]
```

Comment:
Those two interfaces should be exported, they are used in the definition of a public member variable.

Reply:
If I remove those, typedoc throws a warning during the build process. The two types are part of the browser APIs, but not yet in a standard so typescript doesn't know about them.

https://developer.mozilla.org/en-US/docs/Web/API/OffscreenCanvas

Reply:
I would just export the two interfaces to get rid of the warning.

Reply:
I wish it were that easy. (If it is, please send me the fix!)

When I was working with you on the OSMD build, we were having integration issues because we were including the type definitions as a dependency.

I then tried copying the type definitions directly into VexFlow. I ran into issues with that, so I just pared it down to the two interfaces that we reference in VexFlow.

If I export the interfaces, it will generate documentation for them. However, that documentation will be incorrect since my interface definitions (in canvascontext.ts) are incomplete.

These types are built into the browser, so the documentation shouldn't really be generated by us anyways. The API is here: https://developer.mozilla.org/en-US/docs/Web/API/OffscreenCanvas  The spec is here: https://html.spec.whatwg.org/multipage/canvas.html#offscreencanvasrenderingcontext2d

So I'm inclined to leave this as is. Unless you don't like the **intentionallyNotExported** field. If that is the case, I can just remove that field and the two warnings will show up again when geenerating typedoc.

Reply:
See #1297 I have included the API doc changes. This might also reduce the number of files here once you rebase.

Reply:
I pulled your branch and diffed against my current docs. They aren't the same, so it'll cause docs conflicts on any rebase or merge. This PR also includes a typedoc.json which is used to ignore the two types, and adjust the footer.

Reply:
<img width="694" alt="filter" src="https://user-images.githubusercontent.com/239113/150010483-21defbef-db7f-47df-9af2-5068521e9f33.png">

To simplify the review process, you can use the File filter dropdown and uncheck the *.html files.

They were all modified because I added a `typedoc.json` file to contain the options, which includes removing the footer on each page to make it look cleaner.